libjpeg-turbo/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+idf_component_register(
+                       # We need the dummy source file so that the component
+                       # library is not an interface library. This allows to
+                       # get the list of include directories from other components
+                       # via INCLUDE_DIRECTORIES property later on.
+                       SRCS dummy.c)
+
+# Determine compilation flags used for building Jpeg-turbo
+# Flags inherited from IDF build system and other IDF components:
+set(idf_include_directories $<TARGET_PROPERTY:idf::libjpeg-turbo,INCLUDE_DIRECTORIES>)
+set(includes "-I$<JOIN:${idf_include_directories}, -I>")
+
+
+if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug)
+elseif(CONFIG_COMPILER_OPTIMIZATION_SIZE)
+    set(opt_args -DCMAKE_BUILD_TYPE=Relesee)
+elseif(CONFIG_COMPILER_OPTIMIZATION_PERF)
+    set(opt_args -DCMAKE_BUILD_TYPE=Relesee)
+elseif(CONFIG_COMPILER_OPTIMIZATION_NONE)
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug )
+else()
+    message(FATAL_ERROR "Unsupported optimization level")
+endif()
+
+set(extra_defines -D_DEFAULT_SOURCE)
+
+set(c_flags "${includes} ${extra_defines} ")
+set(cxx_flags "${includes} ${extra_defines} ")
+set(common_flags "-ggdb  -ffunction-sections -fdata-sections -fpermissive -lm")
+
+# set(COMPILE_FLAGS "-ggdb  -ffunction-sections -fdata-sections -fpermissive -lm")
+# set(COMPILE_FLAGS "-ggdb  -ffunction-sections -fdata-sections -fpermissive")
+
+if(CONFIG_IDF_TARGET_ARCH_XTENSA)
+    set(assert_flags "${assert_flags} -mlongcalls")
+endif()
+
+if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)
```

Comment:
Looks like this code block is duplicated, you have another `if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)` to set `opt_args`. Could you please clean it up?